### DISCUSSION TITLE
Should knowledge graphs be modeled directly within the biolink model?

### DISCUSSION THREAD

**Discussion by cmungall** [biolink/biolink-model#603] [Ideas]
Our core classes are named thing and association. Should we also have a node to represent a KG?

This is fairly standard in RDF modeling, where the named graph can have its own properties and relationship to other entities, just as nodes and triples can. Some of these may be shared (e.g label, description), others may be specific to graph objects (e.g. aggregate stats like number of edges/nodes). There are various vocabularies for providing information about the graphs (aka metadata). E.g. 

 - https://www.w3.org/TR/hcls-dataset/
 - https://www.w3.org/TR/void/ 
 - https://schema.org/Dataset
 - http://obofoundry.org/ontology/omo

Some of these have partially been aligned with biolink, but this is incomplete

We already have a Distribution class and Dataset, and some minimal metadata around each of these. A KG could be considered a subclass of either of these - or we could simply use the same class.

This would bring "graph metadata" within the realm of biolink data modeling.

I would opt for a very generic definition of KG. It would be inclusive of ontologies. It would include virtual or partial KGs - e.g. results of trapi queries.

**Comment by lpalbou** [comment]
> Should we also have a node to represent a KG
I vote yes !

Even though all data should already be connected across KGs by the model, having a higher level entity could have some useful metadata. I am thinking for instance for a KG containing only predictions (vs experimental); and of the parameters (& date) that were used to create that specific KG
